File: sub-system/src/main/java/org/github/service/impl/ShiroServiceImpl.java
package org.github.service.impl;

import lombok.NonNull;
import lombok.val;
import org.github.auth.feign.IServiceProviderAuthServer;
import org.github.service.IShiroService;
import org.github.service.ISystemService;
import org.github.shiro.AuthorInfo;
import org.github.shiro.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

import static java.util.Objects.requireNonNull;

@Service
public class ShiroServiceImpl implements IShiroService {
    @Autowired
    private IServiceProviderAuthServer authServer;
    @Autowired
    private ISystemService systemService;

    @Override
    public @NonNull List<AuthorInfo> queryAuthorInfo(@NonNull String userId) {
        val token = systemService.token();
        return requireNonNull(authServer.auth(token, userId).getData());
    }

    @Override
    public @NonNull User queryUser(@NonNull String username) {
        val token = systemService.token();
        return requireNonNull(authServer.user(token, username).getData());
    }
}

File: sub-system/src/main/java/org/github/service/impl/ShiroServiceImpl.kt
package org.github.service.impl

import org.github.auth.feign.IServiceProviderAuthServer
import org.github.exception.RemoteErrorException
import org.github.service.IShiroService
import org.github.shiro.AuthorInfo
import org.github.shiro.User
import org.github.spring.restful.json.JSONArrayReturn
import org.github.spring.restful.json.JSONDataReturn
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ShiroServiceImpl: IShiroService {
  @Autowired(required = false)
  private lateinit var authServer: IServiceProviderAuthServer

  override fun queryAuthorInfo(userId: String): JSONArrayReturn<AuthorInfo> {
    val auth = authServer.auth(userId)
    if (auth.failure()) throw RemoteErrorException(auth)
    return auth
  }

  override fun queryUser(username: String): JSONDataReturn<out User> {
    val user = authServer.user(username)
    if (user.failure()) throw RemoteErrorException(user)
    return user
  }
}
